### PR TITLE
Add option for Monday-Friday week

### LINKED ISSUE #417: Add option for Monday-Friday week
Or: TimeTagger wants me to work every weekend.

When setting tag targets, it seems like TimeTagger always defaults to a 7 day work week. If I set daily targets, when the "week" view, I am at only around 71% even if I reached all my daily targets for the week, because TimeTagger insists on counting Saturday and Sunday. If I set weekly targets, the % in the week view is correct, but in the daily view, TimeTagger grossly underestimates the amount of work I need to do each day.

It's not a big deal, but it would be a nice QoL improvement to have a way to get TimeTagger to ignore Saturday and Sunday in target computation.

### PR DESCRIPTION
**Pull request by staddi99**
Closes #417 
Related issue #121 

Added option to set work week to:
- Monday - Friday
- Monday - Saturday
- Monday - Sunday

### CODE REVIEW

File: timetagger/app/front.py [hunk 1]
```diff
@@ -2562,15 +2578,26 @@ def _draw_stats(self, ctx, t1, t2, x1, y1, x2, y2, stat_period, hover):
...
+            not (
+                dt.get_weekday_shortname(t) == "Sat"
+                and window.simplesettings.get("workdays") == 2
+            )
+            and not (
+                dt.get_weekday_shortname(t) == "Sun"
+                and window.simplesettings.get("workdays") >= 1
+            )
+            or not stat_period == "1D"
+        ):
```
[the remark this answers is not in the document]
Comment (staddi99) [reply timetagger/app/front.py:2591]: Will refactor with the efficiency suggestion above and show it also on free days if duration > 0

File: timetagger/app/front.py [hunk 2]
```diff
@@ -3439,7 +3466,12 @@ def _draw_container(self, ctx, total_time, x1, y1, x2, y2):
...
             if best_target:
                 done_this_period = total_time
-                target_this_period = 3600 * best_target.hours * best_target.factor
+                freeDays = dt.get_free_days_in_range(
+                    t1, t2, window.simplesettings.get("workdays")
+                )
+                target_this_period = (
+                    3600 * best_target.hours * (best_target.factor - freeDays)
+                )
                 perc = 100 * done_this_period / target_this_period
```
[the remark this answers is not in the document]
Comment (staddi99) [reply timetagger/app/front.py:3475]: Thank you for bringing this up. This has been implemented to show the correct daily target for periodes greater than a day (week, month, ...).
Will fix the issue if day is dispayed ("day target at Infinity%").

**Comment by staddi99** [comment]
Traget display still to be fixed

File: timetagger/app/front.py [hunk 3]
```diff
@@ -3424,7 +3435,24 @@ def _draw_container(self, ctx, total_time, x1, y1, x2, y2):
...
+                        divisor - window.simplesettings.get("workdays") * 24
+                    )
+                elif period == "month":
+                    factor = cleaned_hours_in_range / (
+                        divisor - window.simplesettings.get("workdays") * 24 * 4.33
+                    )
+                elif period == "year":
+                    factor = cleaned_hours_in_range / (
+                        divisor - window.simplesettings.get("workdays") * 24 * 52
+                    )
```
Comment (almarklein) [inline timetagger/app/front.py:3455]: This PR is looking really good! One final request to make this code easier to follow:
* Please rename `cleaned_hours_in_range` to `work_hours_in_range`.
* Please rewrite the for loop to being `for period in ["day", "week", "month", "year"]:` and replace `divisor` with the actual values.
* Put `free_days_per_week = window.simplesettings.get("workdays")` somewhere, and use that variable instead. Once I figured out that this was what the value means, it all fell into place :)

Sorry to be rather nitpicky but I'm sure someone else will make changes to this code *someday*, and any errors in this logic easily result in subtle bugs that are not obvious at first.

Comment (staddi99) [reply timetagger/app/front.py:3455]: I have also fixed the get_free_hours_in_range function. It now also works for scales smaller than 1 day

**Review by almarklein — COMMENTED** [review]
Very nice work! 🚀

**Pull request merged**